driver-core/src/main/com/mongodb/client/model/GridFSUploadOptions.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+
+/**
+ * GridFS upload options
+ *
+ * Customizable options used when uploading files into GridFS
+ *
+ * @since 3.1
+ */
+public class GridFSUploadOptions {
+    private Integer chunkSizeBytes;
+    private CodecRegistry codecRegistry;
```

Comment:
As I pointed out elsewhere, there should not be a need for the user to provide a codec registry.